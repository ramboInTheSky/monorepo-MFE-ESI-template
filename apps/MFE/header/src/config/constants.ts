export const MODAL_TIMEOUT = 500
export const DEBOUNCE_TIME = 500
export const STICKY_HEADER_SCROLL_TRIGGER = 47

export const LIMIT_AUTO_COMPLETE_SUGGESTIONS = 10

export const BASELINECSS_MATERIALUI = "materialui"
export const BASELINECSS_CUSTOM = "custom"
export const JSS_SERVERSIDE = "header-jss-server-side-"

export const SUPPORTED_QUICKLINK_TYPES = Object.freeze({
    MyAccount: "MyAccount",
    Link: "link",
})

export const SUPPORTED_MY_ACCOUNT_TYPES = Object.freeze({
    loggedIn: "MyAccountLoggedIn",
    loggedOut: "MyAccountLoggedOut",
})

export enum TYPING_STATE {
    AUTO_COMPLETE = "AUTO_COMPLETE",
    RECENT_SEARCH = "RECENT_SEARCH",
    IDLE = "IDLE",
}

export const SITE_URL_HEADER = "x-monorepo-siteurl"
export const REALM_HEADER = "x-monorepo-realm"
export const TERRITORY_HEADER = "x-monorepo-territory"
export const LANGUAGE_HEADER = "x-monorepo-language"
export const SITE_LAYOUT_HEADER = "x-monorepo-viewport-size"
export const DEV_ESI_HEADER = "test-with-local-esi"
export const LOCALHOST = "localhost"

export const AMIDO_REALM = "amido"

export type Anchor = "top" | "bottom" | "left" | "right"

export const MIN_NUMBER_CHAR_TO_SHOW_AUTOCOMPLETE = 3
export const COUNTRY_GB = "gb"
export const REGION_UK = "uk"
export const REGION_INTERNATIONAL = "international"

export const RPID_COOKIE = "RPId"

export const COUNTRY_SELECTOR_DRAWER_BREAKPOINT = "524px"

export const SOLD_OUT = "soldout"
export const IN_STOCK = "instock"
export const QUERY_PARAMETER_TIME_MACHINE_DATE = "time-machine-date"

export const CHEVRON_ICON_URL = "/icons/shared/chevron.svg"
export const CLOSE_ICON_URL = "/icons/shared/close.svg"

export const SOFA_CATEGORY = "sofas"
export const IMAGE = "image"
export const UNDERSCORE = "_"

export const SOFA_CONTENT = "SofaContent"

export const COMMON_ITEM_IMAGE_URL = "/common/items/default/default/itemimages/altitembag/"
export const ITEM_ADD_TO_BAG_URL = "/Item_AddToBag/"
export const JPG = ".jpg"

export const DEFAULT_HEADER_FILE_NAME = "default-header.json"

export const BFPO = "bfpo"
export const ENGLISH_LANGUAGE_CODE = "en"

export const GTM_BLOOMREACH_INFO = "bloomreach"
export const GTM_MONETATE_INFO = "monetateEnabled"
export const GTM_UCM_INFO = "isUserConsentFeatureEnabled"
export const SEARCH_DATA_GA = "Search"
export const SEARCH_DATA_GA_RECENT_SEARCH = "Recent Search"
export const SEARCH_ICON_DATA_GA = "Press Icon"
export const OPEN_SEARCH_DATA_GA = "Open Search"
export const CLOSE_SEARCH_DATA_GA = "Close Search"
export const SEARCH_DELETE_CURRENT_SEARCH_DATA_GA = "Delete Current Search"
export const SEARCH_PREVIOUS_SEARCH_DATA_GA = "Previously Searched"
export const SEARCH_CLEAR_PREVIOUS_SEARCH_DATA_GA = "Clear All Previous Search"
export const SEARCH_KEY_PRESS_ENTER_DATA_GA = "Key Press Enter"
export const HEADER_AUTOCOMPLETE_SEE_ALL_RESULTS_DATA_GA = "See All Results"
export const HEADER_AUTOCOMPLETE_LIST_ITEM_DATA_GA = "list"
export const HEADER_AUTOCOMPLETE_IMAGE_DATA_GA = "image"
export const COUNTRY_LANG_SELECTOR_DATA_GA = "Country Language"
export const COUNTRY_SELECT_DROPDOWN_DATA_GA = "Country Select"

export const SEARCH_BLUR_BREAKPOINT = "768px"

export const HEADER_NAV_BAR_QUICK_LINKS = "Quick Links"
export const HEADER_NAV_BAR_QUICK_LINKS_SHOPPING_BAG = "Shopping bag"
export const HEADER_NAV_BAR_QUICK_LINKS_FAVOURITES = "Favourites"
export const HEADER_NAV_BAR_QUICK_LINKS_MY_ACCOUNT = "My Account"
export const HEADER_NAV_BAR_SHOPPING_BAG_VIEW_BAG = "View Bag"
export const HEADER_NAV_BAR_SHOPPING_BAG_CHECKOUT = "Checkout"
export const HEADER_NAV_BAR_HELP = "Help"
export const HEADER_NAV_BAR_STORE_LOCATOR = "Store Locator"

export const MY_ACCOUNT_URL = "/myaccount"
export const NO_FOLLOW = "nofollow"

export const AMIDO_COOKIE = "AmidoConsentCookie"
export const AMIDO_DIRECT_COOKIE = "AmidoDirectConsentCookie"
export const COOKIE_VALUE = "AllowCookiesFromAmido=True"
export const LOCAL_STORAGE_ACTIVE_DEPT_NAME = "PrimaryNavDeptSelector"
export const VISITED_PAGES = "VisitedPages"

export const BR_COOKIE_COOKIE_PARTS_LENGTH = 4
export const BR_COOKIE_COOKIE_UID_PARTS = 2
export const BR_COOKIE_COOKIE_UID_LENGTH = 2
export const BR_COOKIE_COOKIE_FULL_RANGE = 100

export const BR_COOKIE_COOKIE_NAME = "_br_mt_search"
export const BR_COOKIE_COOKIE_NUMBER_OF_DAYS = 1
export const BR_COOKIE_COOKIE_CACHING_ID = "_br_uid_2"

export const TIME_MACHINE_DATE_COOKIE = "time-machine-date"
export const COUNTRY_SELECTOR_TYPING_DELAY_TIME = 2000

export const RPID = "RPID"
export const ACCOUNT_TYPE = "AcctType"
export const UNLIMITED_CUSTOMER = "unlimitedCustomer"

export const BURGER_MENU_TEMPLATE = "burgerMenu"
export const DEFAULT_TEMPLATE = "default"
export const VARIANT_DEFAULT = "default"

export const PLP_ENTRYPOINT_ID = "plp-entrypoint"
