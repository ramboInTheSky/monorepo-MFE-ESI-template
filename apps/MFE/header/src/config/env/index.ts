const INTERNAL = Object.freeze({
    PRODUCTION: process.env.NODE_ENV === "production",
    DEVELOPMENT: process.env.NODE_ENV === "development",
    DEV_URL_OVERRIDE: process.env.DEV_URL_OVERRIDE || "",
    NODE_ENV: process.env.NODE_ENV,
    REACT_APP_BLOB_STORAGE_PATH: "/static-content",
    USEDEVESI: process.env.USEDEVESI,
})

const ENV_VARS = Object.freeze({
    REACT_APP_BLOB_STORAGE_SSR_BASEURL: process.env.REACT_APP_BLOB_STORAGE_SSR_BASEURL || "http://localhost:3004",
    PORT: process.env.PORT || 3004,
    REACT_APP_CDN_BASEURL: process.env.REACT_APP_CDN_BASEURL
        ? process.env.REACT_APP_CDN_BASEURL
        : "https://xcdn.amido.com/content",
    REACT_APP_API_BASEURL: process.env.REACT_APP_API_BASEURL
        ? process.env.REACT_APP_API_BASEURL
        : "http://sandbox.eun.deploy.systems.next/api/header",
    POD_NAME: process.env.POD_NAME,
    REACT_APP_SERVE_PATH_PREFIX: process.env.REACT_APP_SERVE_PATH_PREFIX || "",
    ASSETS_PATH: process.env.ASSETS_PATH || "/headerstatic",
    LOG_LEVEL: process.env.LOG_LEVEL,
    FNC_NAME: process.env.FNC_NAME,
    REACT_APP_APPINSIGHTS_KEY: process.env.REACT_APP_APPINSIGHTS_KEY,
    BLOOMREACH_BASE_URL: process.env.BLOOMREACH_BASE_URL || "https://brm-suggest-0.brsrvr.com/api/v1/suggest/",
    REACT_APP_MEGANAV_BASEURL:
        process.env.REACT_APP_MEGANAV_BASEURL ||
        (INTERNAL.DEVELOPMENT || INTERNAL.USEDEVESI ? "http://localhost:3005" : ""),
    REACT_APP_MEGANAV_ASSETS_PATH: process.env.REACT_APP_MEGANAV_ASSETS_PATH || "/meganavstatic",
    GEOLOCATION_BASEURL:
        process.env.REACT_APP_GEOLOCATION_BASEURL || "https://services-uat.test.ecomm.systems.next/geolocation",
    ENVIRONMENT: process.env.ENVIRONMENT || "dev",
    REACT_APP_USE_TIME_MACHINE_COOKIE: process.env.REACT_APP_USE_TIME_MACHINE_COOKIE || "false",
    USE_LOCAL_STATIC_CONTENT: process.env.USE_LOCAL_STATIC_CONTENT || "true",
})

const VARS = {...INTERNAL, ...ENV_VARS}

export default VARS
