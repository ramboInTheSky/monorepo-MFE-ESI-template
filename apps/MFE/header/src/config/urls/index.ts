import {formatCdnPath, formatCdnPathWithVariant} from "../../utils/getCdnUrl"

export default Object.freeze({
    searchBox: {
        searchIconUrl: (realm: string, variant): string => formatCdnPathWithVariant("search.svg", realm, variant),
        searchInputButtonUrl: (realm: string, variant): string =>
            formatCdnPathWithVariant("search-input-button.svg", realm, variant),
        clearButton: {
            path: formatCdnPath("/icons/shared/clear.svg"),
        },
    },
    myAccount: {
        loginUrl: "/secure/accounts/transfer",
        signoutUrl: "/forget-me",
        tooltipIconUrl: (realm: string, variant: string): string =>
            formatCdnPathWithVariant("account-tooltip.svg", realm, variant),
    },
    checkout: {
        url: "/secure/checkout/transfer/checkoutcta",
    },
    autoComplete: {
        images: {
            oldDirectoryName: "AltItemZoom",
            newDirectoryName: "search/224x336",
        },
    },
    drawer: {
        iconUrl: formatCdnPath("/icons/shared/amido_close-white.svg"),
    },
    favourites: {
        url: "/favourites",
        activeIconUrl: (realm: string, variant: string) =>
            formatCdnPathWithVariant("favourites-active.svg", realm, variant),
        inactiveIconUrl: (realm: string, variant: string) =>
            formatCdnPathWithVariant("favourites-inactive.svg", realm, variant),
    },
    shoppingBag: {
        iconUrl: (realm: string, variant: string) => formatCdnPathWithVariant("shopping-bag.svg", realm, variant),
        url: "/shoppingbag",
    },
    countryLangSelector: {
        countrySelectUrl: "/countryselect",
    },
    burgerMenu: {
        iconUrl: (realm: string, variant: string) => formatCdnPathWithVariant("menu.svg", realm, variant),
    },
    queueIt: {
        clientUrl: "//static.queue-it.net/script/queueclient.min.js",
        configLoader: "//static.queue-it.net/script/queueconfigloader.js",
    },
    abPlatformTesting: {
        configsPath: "/configs/abplatform/",
        baseFile: "abplatform.js",
        configFile: "abplatformconfig.js",
    },
})
