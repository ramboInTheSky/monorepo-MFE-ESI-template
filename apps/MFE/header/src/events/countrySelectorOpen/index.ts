import {CountrySelectorOpenESB, CountrySelectorOpenContractModel} from "@monorepo/eventservice"

const openESB = new CountrySelectorOpenESB()

const PublishCountrySelectorOpen = (data: CountrySelectorOpenContractModel) => {
    openESB.publish(data)
}
export default PublishCountrySelectorOpen
