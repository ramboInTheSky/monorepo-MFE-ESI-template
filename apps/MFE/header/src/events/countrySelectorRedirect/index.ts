import {CountrySelectorRedirectESB} from "@monorepo/eventservice"

const redirectESB = new CountrySelectorRedirectESB()

const PublishCountrySelectorRedirect = () => {
    redirectESB.publish()
}

export const SubscribeCountrySelectorRedirect = (cb: () => void) => {
    return redirectESB.subscribe(cb)
}

export default PublishCountrySelectorRedirect
