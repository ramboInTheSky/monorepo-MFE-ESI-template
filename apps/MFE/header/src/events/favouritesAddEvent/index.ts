import {AddFavourites, AddFavouritesCallback, AddFavouritesContractModel} from "@monorepo/eventservice"
import {FavouritesApiPost} from "../../api/favourites"
import {getFormattedItemNumber} from "../../utils/favourites"

const addFavouritesESB = new AddFavourites()
const addFavouritesCallbackESB = new AddFavouritesCallback()

export const SubscribeToFavouritesAdd = (siteUrl: string) => {
    addFavouritesESB.subscribe(data => {
        LoadFavouritesApiData(siteUrl, data)
    })
}

class FavouritesAddApiModel {
    itemNumber = 0
    optionCode = ""
    itemUrl = ""
    linkedItemNumber = ""
    linkedOptionNo = ""
    listId = 0
}

const LoadFavouritesApiData = (url: string, data: AddFavouritesContractModel) => {
    const apiModel = new FavouritesAddApiModel()
    apiModel.itemNumber = getFormattedItemNumber(data.itemNumber)
    apiModel.optionCode = data.optionCode
    apiModel.itemUrl = data.itemUrl
    apiModel.linkedItemNumber = data.linkedItemNumber
    apiModel.linkedOptionNo = data.linkedOptionNo
    apiModel.listId = data.listId

    FavouritesApiPost(apiModel, `${url}/favourite`, addFavouritesCallbackESB, data.eventId)
}
