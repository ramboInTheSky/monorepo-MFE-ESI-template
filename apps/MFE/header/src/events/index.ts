/* istanbul ignore file */
export * from "./favouritesGetEvent"
export * from "./favouritesAddEvent"
export * from "./favouritesRemoveEvent"
export * from "./shoppingBagAddEvent"
export * from "./shoppingBagAddCistEvent"
export * from "./shoppingBagGetEvent"
export * from "./shoppingBagAddEVoucherEvent"
export * from "./shoppingBagAddLinkedItemEvent"
export * from "./shoppingBagAddMultipleEvent"
export * from "./shoppingBagAddWarrantyEvent"
export * from "./shoppingBagRemoveEvent"
export * from "./shoppingBagUpdateSizeEvent"
export * from "./shoppingBagUpdateQuantityEvent"
export * from "./modalsClosed"
