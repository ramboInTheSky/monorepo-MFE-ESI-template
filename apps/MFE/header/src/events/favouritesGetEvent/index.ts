import {GetFavourites, GetFavouritesCallback} from "@monorepo/eventservice"
import Cookies from "js-cookie"
import {FavouritesApiGet} from "../../api/favourites"
import {shouldUseSessionStorage, setCacheVersion} from "../../utils/favourites"
import {RPID_COOKIE, MY_ACCOUNT_URL} from "../../config/constants"

const getFavouritesESB = new GetFavourites()
const getFavouritesCallbackESB = new GetFavouritesCallback()

export const SubscribeToFavouritesGet = (siteUrl: string) => {
    getFavouritesESB.subscribe(() => {
        LoadFavouritesApiData(siteUrl)
    })
}

export const LoadFavouritesApiData = (url: string) => {
    const queryString: string = buildQueryString()
    FavouritesApiGet(`${url}/favourite${queryString}`, getFavouritesCallbackESB)
}

const buildQueryString = () => {
    const href = window?.location.href.toLowerCase()
    const isAccountPage = href?.includes(MY_ACCOUNT_URL)
    const shouldCache = () => (isAccountPage ? setCacheVersion(true) : setCacheVersion())
    const cacheVersion = shouldUseSessionStorage() ? shouldCache() : ""
    const rpidCookie = Cookies.get(RPID_COOKIE)
    if (rpidCookie !== undefined) {
        const index = rpidCookie.indexOf("=")
        const rpid = rpidCookie.substring(index + 1)
        return `?id=${rpid}&_=${cacheVersion}`
    }
    return `?_=${cacheVersion}`
}
