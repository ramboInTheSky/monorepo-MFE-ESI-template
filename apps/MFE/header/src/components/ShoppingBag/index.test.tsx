import React from "react"
import {render, screen, fireEvent} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {
    useShoppingBagGetCallbackObservable,
    useShoppingBagAddCallbackObservable,
    useShoppingBagAddCistCallbackObservable,
    useShoppingBagAddEVoucherCallbackObservable,
    useShoppingBagAddLinkedItemCallbackObservable,
    useShoppingBagAddMultipleCallbackObservable,
    useShoppingBagAddWarrantyCallbackObservable,
    useShoppingBagRemoveCallbackObservable,
    useShoppingBagUpdateSizeCallbackObservable,
    useShoppingBagUpdateQuantityCallbackObservable,
    useModalsCloseObservable,
} from "@monorepo/eventservice"
import {PublishToModalsClosed} from "../../events/modalsClosed"
import {ShoppingBag} from "."
import {mockTheme} from "../../../__mocks__/mockStore"
import {LOCAL_STORAGE_ACTIVE_DEPT_NAME} from "../../config/constants"

jest.mock("@monorepo/eventservice")

const mockUpdateShoppingBag = jest.fn()
const mockAddEVoucherToBag = jest.fn()

jest.mock("../../events/modalsClosed")

const props = {
    altText: "photo",
    iconUrl: "/abc123",
    itemCount: 0,
    bag: {ShoppingBag: {Items: []}},
    updateShoppingBag: mockUpdateShoppingBag,
    addEVoucherToBag: mockAddEVoucherToBag,
    isBagLoading: false,
    shoppingBagUrl: "fakeamido.com/shoppingbag",
    miniBagTooltipPlacement: "bottom-end",
}

jest.mock("../MiniShoppingBag")
jest.mock("../ShoppingBagNotification")

const mockSetState = jest.fn()
jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useState: initial => [initial, mockSetState],
}))

describe("Components - ShoppingBag: ", () => {
    beforeEach(() => {
        Object.defineProperty(window, "localStorage", {
            value: {
                getItem: jest.fn(() => null),
                setItem: jest.fn(() => null),
                removeItem: jest.fn(() => null),
            },
            writable: true,
        })
        mockSetState.mockReset()
    })
    it("should match the snapshot", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...props} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it("should call useModalsCloseObservable", () => {
        ;(useModalsCloseObservable as any).mockImplementationOnce(cb => {
            cb("useModalsCloseObservable")
        })
        render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...props} />
            </SCThemeProvider>,
        )
        expect(useModalsCloseObservable).toHaveBeenCalled()
        expect(mockSetState).toHaveBeenCalledWith(false)
    })

    it("should call publishToModalsClosed", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...props} />
            </SCThemeProvider>,
        )
        const link = screen.getByTestId("shopping-bag-link-button")
        fireEvent.click(link)
        expect(PublishToModalsClosed).toHaveBeenCalled()
    })

    it("should match the snapshot when isBagLoading is true", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...{...props, isBagLoading: true}} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it("should subscribe to the ESB events", () => {
        ;(useShoppingBagGetCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagGetCallbackObservable")
        })
        ;(useShoppingBagAddCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddCallbackObservable")
        })
        ;(useShoppingBagAddCistCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddCistCallbackObservable")
        })
        ;(useShoppingBagAddEVoucherCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddEVoucherCallbackObservable")
        })
        ;(useShoppingBagAddLinkedItemCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddLinkedItemCallbackObservable")
        })
        ;(useShoppingBagAddMultipleCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddMultipleCallbackObservable")
        })
        ;(useShoppingBagAddWarrantyCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagAddWarrantyCallbackObservable")
        })
        ;(useShoppingBagRemoveCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagRemoveCallbackObservable")
        })
        ;(useShoppingBagUpdateSizeCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagUpdateSizeCallbackObservable")
        })
        ;(useShoppingBagUpdateQuantityCallbackObservable as any).mockImplementationOnce(cb => {
            cb("useShoppingBagUpdateQuantityCallbackObservable")
        })
        ;(useModalsCloseObservable as any).mockImplementationOnce(cb => {
            cb("useModalsCloseObservable")
        })
        render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...props} />
            </SCThemeProvider>,
        )

        expect(useShoppingBagGetCallbackObservable).toHaveBeenCalled()
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagGetCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagAddCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagAddCistCallbackObservable")
        expect(mockAddEVoucherToBag).toHaveBeenCalledWith("useShoppingBagAddEVoucherCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagAddLinkedItemCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagAddMultipleCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagAddWarrantyCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagRemoveCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagUpdateSizeCallbackObservable")
        expect(mockUpdateShoppingBag).toHaveBeenCalledWith("useShoppingBagUpdateQuantityCallbackObservable")
        expect(useModalsCloseObservable).toHaveBeenCalled()
    })
    it("should remove localStorage when clicked", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <ShoppingBag {...props} />
            </SCThemeProvider>,
        )

        fireEvent.click(screen.getByTestId("header-shopping-bag-link-el"))
        /* eslint-disable */
        expect(window.localStorage.removeItem).toHaveBeenCalledTimes(1)
        expect(window.localStorage.removeItem).toHaveBeenCalledWith(LOCAL_STORAGE_ACTIVE_DEPT_NAME)
        /* eslint-enable */
    })
})
