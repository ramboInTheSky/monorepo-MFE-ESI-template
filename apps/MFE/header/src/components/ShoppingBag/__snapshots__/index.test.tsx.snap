// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - ShoppingBag:  should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 2.75rem;
  line-height: 1.125rem;
}

.c0.c0 i {
  font-style: normal;
  font-size: 0.688rem;
  text-align: center;
  font-family: "AzoSansMedium","Helvetica Neue",Arial,sans-serif;
  font-weight: bold;
  position: absolute;
  width: 100%;
  display: block;
  cursor: pointer;
  color: #fff;
}

.c0.c0 a {
  padding: 0.625rem 0;
}

.c0.c0 button {
  padding: 0.625rem 0.625rem 0.5625rem 0.625rem;
  border: 0;
  background: 0;
}

@media (min-width:1024px) {
  .c0 {
    line-height: initial;
  }
}

@media (min-width:768px) {
  .c0 {
    height: 2.8125rem;
  }
}

@media (min-width:768px) {
  .c0.c0 i {
    font-size: 0.75rem;
  }
}

@media (min-width:1024px) {

}

@media (min-width:768px) {

}

<div
    class="shoppingbag"
    data-testid="header-shopping-bag"
  >
    <div
      class="PrivateHiddenCss-mdUp-8"
    >
      <div
        class="c0"
        data-testid="header-shopping-bag-link-el"
      >
        <a
          data-ga-v1="Quick Links"
          data-ga-v2="Shopping bag"
          href="fakeamido.com/shoppingbag"
          rel="nofollow"
        >
          <img
            alt="photo"
            src="/abc123"
          />
          <i>
            0
          </i>
        </a>
      </div>
    </div>
    <div
      class="PrivateHiddenCss-smDown-6"
    >
      <div
        class="c0"
        data-testid="shopping-bag-link-button"
      >
        <button
          data-ga-v1="Quick Links"
          data-ga-v2="Shopping bag"
          type="button"
        >
          <img
            alt="photo"
            src="/abc123"
          />
          <i>
            0
          </i>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components - ShoppingBag:  should match the snapshot when isBagLoading is true 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 2.75rem;
  line-height: 1.125rem;
}

.c0.c0 i {
  font-style: normal;
  font-size: 0.688rem;
  text-align: center;
  font-family: "AzoSansMedium","Helvetica Neue",Arial,sans-serif;
  font-weight: bold;
  position: absolute;
  width: 100%;
  display: block;
  cursor: pointer;
  color: #fff;
}

.c0.c0 a {
  padding: 0.625rem 0;
}

.c0.c0 button {
  padding: 0.625rem 0.625rem 0.5625rem 0.625rem;
  border: 0;
  background: 0;
}

@media (min-width:1024px) {
  .c0 {
    line-height: initial;
  }
}

@media (min-width:768px) {
  .c0 {
    height: 2.8125rem;
  }
}

@media (min-width:768px) {
  .c0.c0 i {
    font-size: 0.75rem;
  }
}

@media (min-width:1024px) {

}

@media (min-width:768px) {

}

<div
    class="shoppingbag"
    data-testid="header-shopping-bag"
  >
    <div
      class="PrivateHiddenCss-mdUp-53"
    >
      <div
        class="c0"
        data-testid="header-shopping-bag-link-el"
      >
        <a
          data-ga-v1="Quick Links"
          data-ga-v2="Shopping bag"
          href="fakeamido.com/shoppingbag"
          rel="nofollow"
        >
          <img
            alt="photo"
            src="/abc123"
          />
          <i />
        </a>
      </div>
    </div>
    <div
      class="PrivateHiddenCss-smDown-51"
    >
      <div
        class="c0"
        data-testid="shopping-bag-link-button"
      >
        <button
          data-ga-v1="Quick Links"
          data-ga-v2="Shopping bag"
          type="button"
        >
          <img
            alt="photo"
            src="/abc123"
          />
          <i />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
